src/python/pants/engine/internals/parametrize.py

```diff
@@ -166,42 +166,55 @@ def expand(
         except Exception as e:
             raise Exception(f"Failed to parametrize `{address}`:\n  {e}") from e
 
+        parameters = address.parameters
+        non_parametrized = tuple(
+            (field_name, field_value)
+            for field_name, field_value in fields.items()
+            if not isinstance(field_value, Parametrize)
+        )
         if parametrized_groups:
             # Add the groups as one vector for the cross-product.
             parametrized.append(parametrized_groups)
 
+        unparametrize_keys = {k for k, _ in non_parametrized if k in parameters}
```

Comment:
Guess this would mostly be a matter of preference, just curious if there was any particular reasoning motivating the change from:
```python
        unparametrize_keys = non_parametrized & parameters.keys()
```

Reply:
`non_parametrized` is a tuple, so we still need the set comprehension, and using the intersect operand, the whole expression end up being a little longer and less idimatic I found (one comprehension vs one comprehension AND an intersect). But in the end it pretty much boil down to preference.

```
        unparametrize_keys = {k for k, _ in non_parametrized if k in parameters}
        unparametrize_keys = {k for k, _ in non_parametrized} & parameters.keys()
```

Reply:
doh 🤦🏽 😆